src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs

```diff
@@ -0,0 +1,267 @@
+                var registrationParams = JsonConvert.DeserializeObject<RegistrationParams>(@params.ToString());
+                if (registrationParams == null)
+                {
+                    AssertEx.Fail($"Can't parse {@params} to {nameof(RegistrationParams)}");
+                    return;
+                }
+
+                if (registrationParams.Registrations.Any(item => item.Method == Methods.WorkspaceDidChangeConfigurationName))
+                {
+                    if (WorkspaceDidChangeConfigurationRegistered)
+                    {
+                        AssertEx.Fail($"{Methods.WorkspaceDidChangeConfigurationName} is registered twice.");
+                    }
+
+                    WorkspaceDidChangeConfigurationRegistered = true;
+                }
+
+                return;
+            }
+
+            [JsonRpcMethod(Methods.WorkspaceConfigurationName)]
+            public JArray WorkspaceConfigurationName(JToken configurationParamsToken, CancellationToken _)
+            {
+                var configurationParams = JsonConvert.DeserializeObject<ConfigurationParams>(configurationParamsToken.ToString());
+                if (configurationParams == null)
+                {
+                    AssertEx.Fail($"Can't parse {configurationParams} to {nameof(ConfigurationParams)}.");
+                }
+
+                var expectConfigurationItemsNumber = DidChangeConfigurationNotificationHandler.SupportedOptions.Sum(option => option is IPerLanguageValuedOption ? 2 : 1);
+                Assert.Equal(expectConfigurationItemsNumber, configurationParams!.Items.Length);
+                Assert.Equal(expectConfigurationItemsNumber, MockClientSideValues.Count);
+
+                foreach (var item in configurationParams.Items)
+                {
+                    AssertSectionPattern(item.Section);
+                }
+
+                return JArray.FromObject(MockClientSideValues);
+            }
+
+            public void SetClientSideOptionValues(bool setToDefaultValue)
+            {
+                MockClientSideValues.Clear();
+                foreach (var option in DidChangeConfigurationNotificationHandler.SupportedOptions)
+                {
+                    var valueToSet = setToDefaultValue ? GenerateDefaultValue(option) : GenerateNonDefaultValue(option);
+                    if (option is IPerLanguageValuedOption)
+                    {
+                        foreach (var _ in DidChangeConfigurationNotificationHandler.SupportedLanguages)
+                        {
+                            MockClientSideValues.Add(valueToSet);
+                        }
+                    }
+                    else
+                    {
+                        MockClientSideValues.Add(valueToSet);
+                    }
+                }
+            }
+
+            private static string ConvertToString(object? value)
+                => value switch
+                {
+                    null => "null",
```

Comment:
interesting - does the client represent null option values as null?

Reply:
We can define whatever we like (because the protocol define it as LSPAny)
But I think the default behavior is `null` is "null". If the value doesn't exisint, it would be string.Empty.

Reply:
And it's fully controlled by us. It's LSPAny type in the LSP specification.